packages/common/src/utils/replace-null-with-undefined/replace-null-with-undefined.ts

```diff
@@ -0,0 +1,17 @@
+export const replaceNullsWithUndefined = (obj: Record<string, object>) => {
+  if (obj === null) {
+    return undefined;
+  }
+
+  if (typeof obj !== 'object' || Array.isArray(obj)) {
+    return obj;
+  }
+
+  const newObj: any = {};
+  for (const [key, value] of Object.entries(obj)) {
+    newObj[key] =
+      value === null ? undefined : replaceNullsWithUndefined(value as Record<string, object>);
+  }
+
+  return newObj;
+};
```

Comment:
1. Current naming doesn't seem to suggest specificity to objects
2. Working with the `Record` type over the `object` type is usually the intention and is easier to work with in this context
3. Reduce handles for you creating the new object, iteration, and typing
4. Using `unknown` makes sense due to the runtime checks, recursion, and handling `null`
5. Line 12~13 is redundant due to the `obj === null` check at the start of the function

```suggestion
export const replaceNullWithUndefinedInObjects = (obj: unknown) => {
  if (obj === null) {
    return undefined;
  }

  if (typeof obj !== 'object' || Array.isArray(obj)) {
    return obj;
  }

  const objWithoutNulls = Object.entries(obj).reduce((acc, [key, value]) => {
    acc[key] = replaceNullWithUndefinedInObjects(value);

    return acc;
  }, {} as Record<PropertyKey, unknown>);

  return objWithoutNulls;
};
```